typed-racket-test/unit-tests/prefab-tests.rkt

```diff
@@ -20,71 +22,192 @@
+                 3)
+   (check-equal? (prefab-key->field-count
+                  '(foo 1 (1 #f) #() bar 1 (0 #f) #() baz 2 (0 #f) #()))
+                 5)
 
-  (check-abbreviate '(foo) 1)
-  (check-abbreviate '(foo 1) 1)
-  (check-abbreviate '(foo 3) 3)
-  (check-abbreviate '(foo bar 3) 4)
-  (check-abbreviate '(foo 3 bar 3) 6)
-  (check-abbreviate '(foo 1 #()) 1)
-  (check-abbreviate '(foo 1 #(0)) 1)
-  (check-abbreviate '(foo 5 (0 #f)) 5)
-  (check-abbreviate '(foo 5 (1 #f)) 6)
-  (check-abbreviate '(foo 5 (0 #f) #()) 5)
-  (check-abbreviate '(foo 5 (1 #f) #()) 6)
-  (check-abbreviate '(foo 5 (0 #f) #(1)) 5)
-  (check-abbreviate '(foo 5 (1 #f) #(1)) 6)
-  (check-abbreviate '(foo 5 (0 #f) #() bar 1) 6)
-  (check-abbreviate '(foo 5 (0 #f) #(1) bar 1) 6)
-  (check-abbreviate '(foo 5 (1 #f) #(1) bar 1) 7)
-  (check-abbreviate '(foo 5 (1 #f) #(1) bar 1 #()) 7)
-  (check-abbreviate '(foo 5 (0 #f) #(1) bar 1 #()) 6)
-  (check-abbreviate '(foo 5 (0 #f) #() bar 1 #()) 6)
-  (check-abbreviate '(foo 5 (1 #f) #(1) bar 1 #(0)) 7)
+   (check-abbreviate '(foo) 1)
+   (check-abbreviate '(foo 1) 1)
+   (check-abbreviate '(foo 3) 3)
+   (check-abbreviate '(foo bar 3) 4)
+   (check-abbreviate '(foo 3 bar 3) 6)
+   (check-abbreviate '(foo 1 #()) 1)
+   (check-abbreviate '(foo 1 #(0)) 1)
+   (check-abbreviate '(foo 5 (0 #f)) 5)
+   (check-abbreviate '(foo 5 (1 #f)) 6)
+   (check-abbreviate '(foo 5 (0 #f) #()) 5)
+   (check-abbreviate '(foo 5 (1 #f) #()) 6)
+   (check-abbreviate '(foo 5 (0 #f) #(1)) 5)
+   (check-abbreviate '(foo 5 (1 #f) #(1)) 6)
+   (check-abbreviate '(foo 5 (0 #f) #() bar 1) 6)
+   (check-abbreviate '(foo 5 (0 #f) #(1) bar 1) 6)
+   (check-abbreviate '(foo 5 (1 #f) #(1) bar 1) 7)
+   (check-abbreviate '(foo 5 (1 #f) #(1) bar 1 #()) 7)
+   (check-abbreviate '(foo 5 (0 #f) #(1) bar 1 #()) 6)
+   (check-abbreviate '(foo 5 (0 #f) #() bar 1 #()) 6)
+   (check-abbreviate '(foo 5 (1 #f) #(1) bar 1 #(0)) 7)
 
-  (check-true (prefab-key-subtype? '(foo 1 (0 #f) #() bar 1 (0 #f) #())
-                                   '(bar 1 (0 #f) #())))
-  (check-false (prefab-key-subtype? '(foo 1 (0 #f) #())
+   (check-true (prefab-key-subtype? '(foo 1 (0 #f) #() bar 1 (0 #f) #())
                                     '(bar 1 (0 #f) #())))
+   (check-false (prefab-key-subtype? '(foo 1 (0 #f) #())
+                                     '(bar 1 (0 #f) #())))
+
+   (check-equal? (prefab-key->field-mutability
+                  '(foo 1 (0 #f) #() bar 1 (0 #f) #()))
+                 (list #f #f))
+   (check-equal? (prefab-key->field-mutability
+                  '(foo 2 (0 #f) #(0) bar 3 (0 #f) #(1)))
+                 (list #f #t #f #t #f))
+   (check-equal? (prefab-key->field-mutability
+                  '(foo 2 (0 #f) #(0) bar 3 (0 #f) #(1) baz 1 (0 #f) #(0)))
+                 (list #t #f #t #f #t #f))
+   ;; immutable prefab example
+   (check-eq? raw-p p0)
+   (check-equal? (pair-fst p0) 1)
+   (check-equal? (pair-snd p0) 2)
+   (check-exn exn:fail:contract:blame?
```

Comment:
These should probably use something like the infrastructure in the contract test suite to check that the blame is right.